src/Space.h

```diff
@@ -106,11 +106,11 @@ class Space {
 	Game *m_game;
 
 	// all the bodies we know about
-	std::list<Body *> m_bodies;
+	std::vector<Body *> m_bodies;
 
 	// bodies that were removed/killed this timestep and need pruning at the end
-	std::list<Body *> m_removeBodies;
-	std::list<Body *> m_killBodies;
+	std::vector<Body *> m_removeBodies;
+	std::vector<Body *> m_killBodies;
```

Comment:
I'd recommend removing these two and replacing them with a single `std::vector<std::pair<Body *, uint32_t>>` list, and storing whether the body should be removed or killed as a flag in that integer. IIRC not many bodies actually need to go on the remove-but-not-kill list (possibly the other way around), and for efficiency reasons it's much more performant to only have one list of bodies to remove.